src/lib/api/__tests__/genres.test.ts

```diff
@@ -0,0 +1,137 @@
+import { describe, it, expect, beforeEach, beforeAll, afterEach, afterAll } from 'vitest';
+import { http, HttpResponse } from 'msw';
+import { setupServer } from 'msw/node';
+import { getGenres, clearGenresCache } from '../genres';
+import { API_URL } from '@/constants';
+
+const handlers = [
+  http.get(`${API_URL}/genres`, () => {
+    return HttpResponse.json([
+      'Rock',
+      'Jazz',
+      'Classical',
+      'Electronic',
+      'Hip-Hop',
+      'Pop'
+    ])
+  }),
+];
+
+const server = setupServer(...handlers);
+
+beforeAll(() => {
+  server.listen({ onUnhandledRequest: 'error' });
+});
+
+afterEach(() => {
+  server.resetHandlers();
+  clearGenresCache();
+});
+
+afterAll(() => {
+  server.close();
+});
+
+describe('getGenres Integration Tests', () => {
+  it('should successfully fetch and return genres from the API', async () => {
+    const result = await getGenres();
+    
+    expect(result.isOk()).toBe(true);
+    if (result.isOk()) {
```

Comment:
Do we need an if statement here? Wouldn't the test fail anyway, if result.isOk is not truthy?

Same for other tests